src/Injector/AbstractInjector.php

```diff
@@ -162,6 +181,155 @@ public function inject($package, $type, IOInterface $io)
     }
 
     /**
+     * Inject component $package after all dependencies into $config and return true.
+     * If any of dependencies is not registered the method will write error to $io
+     * and will also return true, to prevent injecting this package later.
+     * Method return false only in case when dependencies for the package are not found.
+     *
+     * @param string $package
+     * @param string $config
+     * @param IOInterface $io
+     * @return bool
+     */
+    private function injectAfterDependencies($package, $config, IOInterface $io)
+    {
+        $moduleClass = sprintf('%s\\%s', $package, 'Module');
+        if (class_exists($moduleClass) && method_exists($moduleClass, 'getModuleDependencies')) {
+            $module = new $moduleClass();
+            $dependencies = $module->getModuleDependencies();
+
+            if ($dependencies) {
+                foreach ($dependencies as $dependency) {
+                    if (! $this->isRegisteredInConfig($dependency, $config)) {
+                        $io->write(sprintf(
+                            '<error>    Dependency %s is not registered in the configuration</error>',
+                            $dependency
+                        ));
+
+                        return true;
```

Comment:
This is new functionality, so we can see how it plays out as people start updating to the 0.4 series. My gut feeling is that we report the problem, and continue on injecting other dependencies. The user sees the error, and should then realize there may be a problem — e.g., a package depends explicitly on another within the code, but does not declare it in its own Composer configuration. In such cases, they can then report the problem upstream.